File: TaskMaster/TaskMaster/Models/Task.cs
namespace TaskMaster.Models
{
    public class Task
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime DueDate { get; set; }
        public bool IsCompleted { get; set; }
        public Priority Priority { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    }

    public enum Priority
    {
        Low,
        Medium,
        High
    }
}

File: TaskMaster/frontend/src/models/Task.ts
export interface Task {
    id: number;
    title: string;
    description: string;
    dueDate: Date;
    isCompleted: boolean;
    priority: Priority;
    createdAt: Date;
}

export interface CreateTaskRequest {
    title: string;
    description: string;
    dueDate?: Date;
    priority: Priority; 
}

export enum Priority {
    Low = 0,
    Medium = 1,
    High = 2
}